serial.go

```diff
@@ -84,12 +89,36 @@ type Config struct {
 	// XONFlowControl bool
 
 	// CRLFTranslate bool
-	// TimeoutStuff int
 }
 
 // OpenPort opens a serial port with the specified configuration
 func OpenPort(c *Config) (io.ReadWriteCloser, error) {
-	return openPort(c.Name, c.Baud)
+	return openPort(c.Name, c.Baud, c.ReadTimeout)
+}
+
+// Converts the timeout values for Linux / POSIX systems
+func posixTimeoutValues(readTimeout time.Duration) (vmin uint8, vtime uint8) {
+	const MAXUINT8 = 1<<8 - 1 // 255
+	// set blocking / non-blocking read
+	var minBytesToRead uint8 = 1
+	var readTimeoutInDeci uint8 = 0
+	if readTimeout > 0 {
+		// EOF on zero read
+		minBytesToRead = 0
+		timeoutMs := uint32(readTimeout.Nanoseconds() / 1e6)
```

Comment:
It seems like this could be simpler with something like

```
var readTimeoutInDeci int64
if readTimeout > 0 {
    readTimeoutInDeci = readTimeout.Nanoseconds() / 1e6 / 100
    if readTimeoutInDeci < 1 {
        readTimeoutInDeci = 1
    }
    if readTimeoutInDeci > MAXUINT8 {
        readTimeoutInDeci = MAXUINT8
    }
}

vtime = uint8(readTimeoutInDeci)
```